spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1347,5 +1352,15 @@ Emitter<String> emitter;
 ----
 
 Since the `@Channel("myChannel")` is used to produce messages, a consumer with the `@Incoming("myChannel")` should be specified to consume the messages transiting on the channel _myChannel_.
-An `IllegalStateException` will be thrown if no consumer is found by the time a message is emitted to the channel.
+
+== Assembly and validation
+
+When the application starts, the Reactive Messaging implementation:
+
+* connect the methods annotated with `@Incoming` and `@Outgoing,` `Emitters,` channels and connectors
+* verify the validity of the resulting graph
+
+If the implementation cannot find a matching _upstream_ or _downstream_ channel for one of the components, a `DeploymentException` must be thrown when the application starts.
+The implementation must also throw a `DeploymentException` if components are illegally connected to multiple upstreams or downstreams.
+If the implementation detects a missing connector, a `DeploymentException` must also be thrown.
```

Comment:
How about updating the previous 3 sentences with this:
A `DeploymentException` must be thrown for any of the following conditions:
* mismatching _upstream_ and _downstream_ channels 
* multiple _upstream_ or _downstream_ specified
* missing connector

Reply:
I extended the list with all the cases.